amazon/ionbenchmark/benchmark_runner.py

```diff
@@ -114,21 +114,51 @@ def test_fn():
         data_file = benchmark_spec.get_input_file()
 
         def test_fn():
-            with open(data_file, "rb") as f:
-                return loader_dumper.load(f)
+            format_option = benchmark_spec.get_format()
+            if _format.format_is_ion(format_option):
```

Comment:
Put the loader specific logic in custom loaders for each type, avoid branching here.